src/lightning_app/core/app.py

```diff
@@ -352,17 +356,24 @@ def run_once(self):
 
         _process_command_requests(self)
 
+        t0 = time()
+
         try:
             self.check_error_queue()
-            t0 = time()
-            self.root.run()
-            self._last_run_time = time() - t0
+            # Execute the flow only if:
+            # - There are state changes
+            # - It is the first execution of the flow
+            if self._has_updated or not self.first_execution:
+                self.root.run()
         except CacheMissException:
             self._on_cache_miss_exception()
         except (ExitAppException, KeyboardInterrupt):
             done = True
             self.stage = AppStage.STOPPING
 
+        self._last_run_time = time() - t0
+        self.first_execution = True
```

Comment:
Why is this a public attribute and why do we need it? Wouldn't it be sufficient set `self._has_updated = True` in the beginning. If running is defined by state updates, we could say that the very first run is triggered by a state update.